### ISSUE TITLE
Error: The 'Access-Control-Allow-Origin' header contains multiple values '*, *', but only one is allowed.

### ISSUE THREAD

**Issue by biapar** [mattbrailsford/umbraco-authu#16]
If you enable CORS in WEB API,you have the error  after OPTIONS request
```

    var cors = new EnableCorsAttribute("*", "*", "*");
            config.EnableCors(cors);
```
Error:
**The 'Access-Control-Allow-Origin' header contains multiple values '*, *', but only one is allowed.**

If you don't enable the CORS, you had the error 405 to chck OPTIONS

To solve the problem, I comment out the function **SetAllowedOriginHeader** into the function  
`public object Token(OAuthTokenRequest request)`
in the file OAuthController

**Comment by craigriviere** [comment]
I've spent ages looking for reason as to why this doesn't work and finally someone with the exact same problem.

Could I ask how do you override the internal controllers function, as I don't have access to those files, seen as it was included in my project via NuGet

**Comment by biapar** [comment]
This problem afflicting the javascript/angular scripts.
Another workaround is to create an overloading of WebApiConfig and put in:

`public static class WebApiConfig
    {

        public static void Register(HttpConfiguration config)
        {

            //var cors = new EnableCorsAttribute("*", "*", "*");
            //config.EnableCors(cors);
            config.EnableCors();



        }

    }`

and then

`protected override void ApplicationInitialized(UmbracoApplicationBase umbracoApplication, ApplicationContext applicationContext)
        {
            base.ApplicationInitialized(umbracoApplication, applicationContext);

            GlobalConfiguration.Configure(WebApiConfig.Register);
        }`

into a class that override Umbraco ApplicationEventHandler

**Comment by biapar** [comment]
Check thread update.

________________________________
Da: Craig <notifications@github.com>
Inviato: giovedì 4 gennaio 2018 18:54:45
A: mattbrailsford/umbraco-authu
Cc: BP; Author
Oggetto: Re: [mattbrailsford/umbraco-authu] Error: The 'Access-Control-Allow-Origin' header contains multiple values '*, *', but only one is allowed. (#16)


I've spent ages looking for reason as to why this doesn't work and finally someone with the exact same problem.

Could I ask how do you override the internal controllers function, as I don't have access to those files, seen as it was included in my project via NuGet

**Comment by ghost** [comment]
+1, would be nice if CORS header manipulation is disabled when `AllowedOrigin` is omitted. Either that or make it configurable during the initialization with another property. Would make life easier when developing single page apps

**Comment by JohnBergman** [comment]
This might be of some help for you.  https://our.umbraco.org/forum/umbraco-7/using-umbraco-7/63841-Web-API-2-and-cross-origin-resource-sharing-CORS

**Comment by mattbrailsford** [comment]
Anyone want to have a stab at a PR for this? Should probably do some kind of check to see if CORS has been enabled some place else and then not bother doing it within AUTHU

**Comment by biapar** [comment]
@mattbrailsford I found the workaround as I wrote in the my first comment, but I've not open a PR.

**Comment by mattbrailsford** [comment]
Ok, I tried to debug this and have the following setup

![image](https://user-images.githubusercontent.com/527305/44835642-b92be400-ac2c-11e8-900d-1ffe2f2da9dc.png)

However this seems to work without issue for me. I can authenticate and I can then access by custom API.

Under what conditions does this error occur?

**Comment by mattbrailsford** [comment]
Ok, I managed to replicate it now. I tried to see if there was a way to check if cors had been confirgued, but it was going to get messy, so the best I can think of is for you to need to eplicitly set `AllowedOrigin` to null in the `OAuthOptions` (still defaulting to `*` if you don't set it).

I think this is ok as it explicitly requires you to opt out in this way. Thoughts?

![image](https://user-images.githubusercontent.com/527305/44837711-3d349a80-ac32-11e8-8eee-02a204b84e28.png)
